File: problems/algorithms/object-order-item/ruby/response/print/products_qtd.print.rb
require_relative "../src/products_qtd"

products = [
    {
      id: 1,
      preco: 10.0,
      qtd: 2,
    },
    {
      id: 2,
      preco: 10.0,
      qtd: 2,
    },
    {
      id: 3,
      preco: 10.0,
      qtd: 2,
    },
    {
      id: 4,
      preco: 10.0,
      qtd: 0,
    },
  ]
  
  # Cart Tool
  
  # filtering greater than zero
  puts Products.gt_zero(products))
  puts [
    {
      id: 1,
      preco: 10.0,
      qtd: 2,
    },
    {
      id: 2,
      preco: 10.0,
      qtd: 2,
    },
    {
      id: 3,
      preco: 10.0,
      qtd: 2,
    },
  ]
  
  # calculating subtotal
  puts Products.sub_total(products))
  puts [
    {
      id: 1,
      subtotal: 20.0,
    },
    {
      id: 2,
      subtotal: 20.0,
    },
    {
      id: 3,
      subtotal: 20.0,
    },
    {
      id: 4,
      subtotal: 0.0,
    },
  ]
  
  # calculating total
  puts Products.total(products))
  puts 60.0
  

File: src/pages/problems/array-products-qtd/_codes/js/response/products-qtd.print.js
import { gtZero, subTotal, total } from "./products-qtd.js";

// Cart Tool
const products = [
  [1, 10.0, 2],
  [2, 10.0, 2],
  [3, 10.0, 2],
  [4, 10.0, 0]
];

// filtering greater than zero
console.log(gtZero(products));
console.log([
  [1, 10.0, 2],
  [2, 10.0, 2],
  [3, 10.0, 2]
]);

// calculating subtotal
console.log(subTotal(products));
console.log([
  [1, 20.0],
  [2, 20.0],
  [3, 20.0],
  [4, 0.0]
]);

// calculating total
console.log(total(products));
console.log(60.0);
